libnavui-androidauto/src/main/java/com/mapbox/androidauto/internal/car/search/CarPlaceSearchImpl.kt

```diff
@@ -0,0 +1,134 @@
+package com.mapbox.androidauto.internal.car.search
+
+import com.mapbox.androidauto.car.search.CarPlaceSearch
+import com.mapbox.androidauto.internal.logAndroidAuto
+import com.mapbox.androidauto.internal.logAndroidAutoFailure
+import com.mapbox.navigation.base.ExperimentalPreviewMapboxNavigationAPI
+import com.mapbox.navigation.core.MapboxNavigation
+import com.mapbox.search.MapboxSearchSdk
+import com.mapbox.search.ResponseInfo
+import com.mapbox.search.SearchEngine
+import com.mapbox.search.SearchEngineSettings
+import com.mapbox.search.SearchOptions
+import com.mapbox.search.SearchSelectionCallback
+import com.mapbox.search.SearchSuggestionsCallback
+import com.mapbox.search.result.SearchResult
+import com.mapbox.search.result.SearchSuggestion
+import kotlinx.coroutines.suspendCancellableCoroutine
+import kotlin.coroutines.resume
+
+/**
+ * Default implementation of the [CarPlaceSearch].
+ */
+@OptIn(ExperimentalPreviewMapboxNavigationAPI::class)
+class CarPlaceSearchImpl(
+    private val searchEngineSettings: SearchEngineSettings? = null,
+) : CarPlaceSearch {
+
+    private var searchEngine: SearchEngine? = null
+    private val locationProvider = CarSearchLocationProvider()
+
+    override fun onAttached(mapboxNavigation: MapboxNavigation) {
+        locationProvider.onAttached(mapboxNavigation)
+        val settings = searchEngineSettings
+            ?: SearchEngineSettings(
+                accessToken = mapboxNavigation.navigationOptions.accessToken!!,
+                locationProvider,
+            )
+        searchEngine = MapboxSearchSdk.createSearchEngineWithBuiltInDataProviders(settings)
```

Comment:
```suggestion
        searchEngine = MapboxSearchSdk.createSearchEngineWithBuiltInDataProviders(ApiType.SBS, settings)
```
Or let users pass `ApiType` from the outside.

Reply:
I don't think we want to pass `ApiType` from the outside, specifically for place search. The search screen is only using the "Single Box Search API."

Also, it is experimental and we're making a stable api https://github.com/mapbox/navigation-sdks/issues/1904